controllers/category.js

```diff
@@ -115,12 +102,27 @@ exports.update = async (req, res) => {
       res.status(422).json({ errors: errors.array() });
       return;
     }
+
+    let imagesArray = [];

```

Comment:
what happened if user doesn't want to update images?

Reply:
let payload = {
        categoryName,
        subCategories,
}
if(req.files.length){
    let imagesArray = [];
    req.files.forEach((element) => {
      const file = {
        imageName: element.originalname,
        imagePath: element.path,
        imageType: element.mimetype,
        fileSize: fileSizeFormatter(element.size, 2),
      };
      imagesArray.push(file);
    });
    payload.categoryImages = imagesArray
}
const updatedCartegory = await Category.findByIdAndUpdate(req.params.id, {
      $set: payload,
      { new: true }
    );
    
    Try logic something like that